packages/app-desktop/gui/ResizableLayout/utils/useLayoutItemSizes.test.ts

```diff
@@ -138,4 +138,157 @@ describe('useLayoutItemSizes', () => {
 		expect(itemSize(parent.children[1], parent, sizes, false)).toEqual({ width: 95, height: 50 });
 	});
 
+	test('should decrease size of the largest item if the total size would be larger than the container', () => {
+		const layout: LayoutItem = validateLayout({
+			key: 'root',
+			width: 200,
+			height: 100,
+			direction: LayoutItemDirection.Row,
+			children: [
+				{
+					key: 'col1',
+					width: 110,
+				},
+				{
+					key: 'col2',
+					width: 100,
+				},
+				{
+					key: 'col3',
+					minWidth: 50,
+				},
+			],
+		});
+
+		const { result } = renderHook(() => useLayoutItemSizes(layout));
+		const sizes = result.current;
+
+		expect(sizes.col1.width).toBe(50);
+		expect(sizes.col2.width).toBe(100);
+	});
+
+	test('should ignore invisible items when counting remaining size', () => {
+		const layout: LayoutItem = validateLayout({
+			key: 'root',
+			width: 200,
+			height: 100,
+			direction: LayoutItemDirection.Row,
+			children: [
+				{
+					key: 'col1',
+					width: 110,
+					visible: false,
+				},
+				{
+					key: 'col2',
+					width: 100,
+				},
+				{
+					key: 'col3',
+					minWidth: 50,
+				},
+			],
+		});
+
+		const { result } = renderHook(() => useLayoutItemSizes(layout));
+		const sizes = result.current;
+
+		expect(sizes.col2.width).toBe(100);
```

Comment:
I'm not sure what this test demonstrates. Does it imply that col3 width is 100? If so could you add a check for col3 too please?